### PR TITLE
Updating Travis to enforce branch strategy

### PR DESCRIPTION
**Pull request by marstr**
Adding this clause should add a task that must pass, which checks to see if CI is running for a PR against master in the private repository.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -66,6 +67,10 @@ script:
     if [[ $MODE == 'java' ]]; then
       $DOCKER_CMD Azure/azure-libraries-for-java -v
     fi
+  - >-
+    if [[ $MODE == 'branchStrategy']]; then
+      [[ $TRAVIS_PULL_REQUEST && $TRAVIS_PULL_REQUEST == 'Azure/azure-rest-api-specs-pr' && $TRAVIS_BRANCH == 'master' ]]
```
Comment (dsgouda) [inline .travis.yml]: nit: Since I am new to this, should you also check `$TRAVIS_BRANCH` for null?

**Review by dsgouda — APPROVED** [review]
LGTM, feel free to merge

**Comment by marstr** [comment]
CI must pass here, and CI must fail in Azure/azure-rest-api-specs-pr#430 must fail.

File: .travis.yml (hunk 1 above)
Comment (marstr) [reply .travis.yml]: Relevant stuff from here: https://docs.travis-ci.com/user/environment-variables/

> TRAVIS_BRANCH:
for push builds, or builds not triggered by a pull request, this is the name of the branch.
for builds triggered by a pull request this is the name of the branch targeted by the pull request.
for builds triggered by a tag, this is the same as the name of the tag (TRAVIS_TAG).

> TRAVIS_PULL_REQUEST: The pull request number if the current job is a pull request, “false” if it’s not a pull request.

> TRAVIS_PULL_REQUEST_SLUG:
if the current job is a pull request, the slug (in the form owner_name/repo_name) of the repository from which the PR originated.
if the current job is a push build, this variable is empty ("").

> TRAVIS_REPO_SLUG: The slug (in form: owner_name/repo_name) of the repository currently being built.

Looks like I do have a typo though! For the second clause, I meant to have it be `$TRAVIS_PULL_REQUEST_SLUG` not `$TRAVIS_PULL_REQUEST`. Looking at it again though, I'll just use `$TRAVIS_REPO_SLUG`.

**Review by olydis — APPROVED** [review]
there we go 👍

**Pull request merged**